main.cpp

```diff
@@ -16,19 +20,48 @@ std::ostream &operator<<(std::ostream &os, std::vector<T> const &a) {
 
 // 请修复这个函数的定义：10 分
 template <class T1, class T2>
-std::vector<T0> operator+(std::vector<T1> const &a, std::vector<T2> const &b) {
+decltype(auto) operator+(std::vector<T1> const &a, std::vector<T2> const &b) {
     // 请实现列表的逐元素加法！10 分
     // 例如 {1, 2} + {3, 4} = {4, 6}
+    using T0 = decltype(T1{} + T2{});
+    std::vector<T0> res;
+    for(size_t i=0; i<a.size()&&i<b.size(); i++){
+        res.push_back(a[i]+b[i]);
+    }
+    return res;
 }
 
 template <class T1, class T2>
-std::variant<T1, T2> operator+(std::variant<T1, T2> const &a, std::variant<T1, T2> const &b) {
+decltype(auto) operator+(std::variant<T1, T2> const &a, std::variant<T1, T2> const &b) {
```

Comment:
```suggestion
auto operator+(std::variant<T1, T2> const &a, std::variant<T1, T2> const &b) {
```
`decltype(auto)` 是用来自动匹配引用类型的，建议用 `auto`，因为这里只需要匹配值类型。